pilot/pkg/util/network/ip.go

```diff
@@ -144,16 +144,38 @@ func ResolveAddr(addr string, lookupIPAddr ...lookupIPAddrType) (string, error)
 // IsIPv6Proxy check the addresses slice and returns true for all addresses are valid IPv6 address
 // for all other cases it returns false
 func IsIPv6Proxy(ipAddrs []string) bool {
+	result := false
 	for i := 0; i < len(ipAddrs); i++ {
 		addr := net.ParseIP(ipAddrs[i])
 		if addr == nil {
 			// Should not happen, invalid IP in proxy's IPAddresses slice should have been caught earlier,
 			// skip it to prevent a panic.
 			continue
 		}
+
+		if addr.To4() == nil && addr.To16() != nil && !addr.IsLinkLocalUnicast() {
```

Comment:
It's hard to remember the reason, but I believe it because a proxy can have an IPv6 address (CRI supports ipv6) but you haven't configured K8s for dual-stack support. In that case an ipv6 link local address will appear, but not one that is routable to with K8s.

Reply:
If only has LinkLocalUnicast address, it doesn't treat it as IPv6Proxy since it won't be able to handle traffic on that address.

Reply:
okay, I think I can comment this in code

Reply:
done